controllers/__init__.py

```diff
@@ -47,14 +48,25 @@ def bootstrap_indexes(self) -> None:
                     f"Created DB index {keys}, collection: {collection}"
                 )
 
-    def get_circuits(self, archived: Optional[bool] = False) -> Dict:
+    def get_circuits(self, archived: Optional[bool] = False,
+                     metadata: dict = None) -> Dict:
         """Get all circuits from database."""
         aggregation = []
+        options = {"null": None, "true": True, "false": False}
         match_filters = {"$match": {}}
+        aggregation.append(match_filters)
         if archived is not None:
+            archived = options.get(archived, False)
             match_filters["$match"]["archived"] = archived
-            aggregation.append(match_filters)
-
+        if metadata:
+            for key in metadata:
+                if "metadata." in key[:9]:
+                    try:
+                        match_filters["$match"][key] = int(metadata[key])
+                    except ValueError:
+                        item = metadata[key].lower()
+                        match_filters["$match"][key] = options.get(item, item)
+        print("AGREE -> ", aggregation)
```

Comment:
```suggestion
```

Can you remove this print leftover here?

Reply:
Deleted.